### PR TITLE
Bookmarks + Playback: Add handling for when both the bookmarks what's new and the playback prompt will show

### PR DESCRIPTION
**Pull request by emilylaguna**
With both the Bookmarks and Playback features being enabled in the same version, we are presented with the problem of both prompts trying to be shown at the _same_ time  which results only one being shown, and an error being displayed in console about double presenting. 💥💥

To fix this, I've updated the logic so that:
- If the What's New is displayed the Playback Prompt will not be
   - If the user dismisses the announcement (maybe later button), the Playback Prompt will be shown if the user is eligible 
   - If the user uses the announcements action button (blue button), the Playback prompt will not be shown until the next launch. At which point only the Playback prompt will be displayed
- If the user opens the Playback Stories without seeing the prompt, the prompt will no longer display on the next launch
- If the what's new is not displayed, the Playback prompt will show as it did before.

I've also added a flag to prevent the What's New or Playback Prompt display logic from being ran more than once per app lifecycle. The logic is called in `MainTabBarController.viewDidAppear` which can be fired more than once, such as if the Playback Stories are shown and then dismissed. 

## To test
There's a lot of steps, but the goal is to test all the states, and the transition between states (logged out to logged in). 

**Before running do the following:**
   1. Enable the End of Year feature flag
   2. Change the 7.52 references in Announcements.swift to 7.51
   3. Change BuildEnvironment.swift line 24 from `.debug` to `.testFlight`

#### Logged out user
1. Perform a fresh install of the app
2. ✅ Verify you see the initial onboarding prompt
3. ✅ Verify you do not see the what's new or playback prompt

#### Logged out user → Signs In → Not Eligible for Playback or What's New
1. Perform a fresh install of the app
2. ✅ Verify you see the initial onboarding prompt
3. Sign into or create an account that is not eligible for playback or has plus
4. ✅ Verify you do not see the what's new or playback prompt after signing in 

#### Logged out user → Signs In → Eligible for Playback, Ineligible for What's New
1. Perform a fresh install of the app
2. ✅ Verify you see the initial onboarding prompt
3. Sign into an account that is eligible for playback and does not have plus
4. ✅ Verify you see the Playback Prompt
5. ✅ Verify you see the Playback section in Profile

#### Logged out user → Signs In → Eligible for Playback and What's New
1. Perform a fresh install of the app
2. ✅ Verify you see the initial onboarding prompt
3. Sign into an account that is eligible for playback and has Plus
4. ✅ Verify you see the Playback Prompt
5. ✅ Verify you see the Playback section in Profile
6. Relaunch the app
9. ✅ Verify you see no prompt

#### Logged out user → Signs In → Eligible for What's New, not for Playback
1. Perform a fresh install of the app
2. ✅ Verify you see the initial onboarding prompt
3. Sign into an account that is not eligible for playback and has plus
4. ✅ Verify you see no prompts after signing in
5. Relaunch the app

#### Logged out user → Becomes eligible for Playback
1. Perform a fresh install of the app
2. Dismiss the login
3. Listen to at least 5 minutes of an episode
4. Go to the Profile tab
5. ✅ Verify you see the Playback section
6. Relaunch the app
7. ✅ Verify you see the Playback Prompt

#### Logged out user → Becomes eligible for Playback → Open Stories without seeing prompt
1. Perform a fresh install of the app
2. Dismiss the login
3. Listen to at least 5 minutes of an episode
4. Go to the Profile tab
6. Tap the Playback section
7. Sign into an account or create an account
8. ✅ Verify you see the Playback stories
9. Dismiss it
10. Relaunch the app
11. ✅ Verify you do not see the Playback Prompt

#### Logged In → Eligible for What's New and for Playback → Dismisses What's New
1. Launch the app from a previous version that is signed into an account eligible for Playback and has Plus
2. ✅ Verify you see the What's New
3. Tap the Maybe Later button
4. ✅ Verify you see the Playback Prompt

#### Logged In → Eligible for What's New and for Playback → What's New Action
1. Launch the app from a previous version that is signed into an account eligible for Playback and has Plus
2. ✅ Verify you see the What's New
3. Tap the action button
4. ✅ Verify the action is performed
5. ✅ Verify you don't see the Playback prompt
6. Relaunch the app
7. ✅ Verify you see the Playback Prompt

#### Logged In → Eligible for Playback
1. Launch the app from a previous version that is signed into an account eligible for Playback and does not have Plus
2. ✅ Verify you see the Playback prompt
6. Relaunch the app
7. ✅ Verify you see nothing

#### Logged out → Eligible for Playback → Signs in
1. Launch the app from a previous version that is signed into an account eligible for Playback and does not have Plus
2. ✅ Verify you see the Playback prompt
6. Tap the View button
8. ✅ Verify you are prompted to sign in
9. Sign in
10. ✅ Verify you see the Playback Stories
11. Dismiss it
12. ✅ Verify you do not see anything
13. Relaunch the app
14. ✅ Verify you see nothing

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

### CODE REVIEW

File: podcasts/MainTabBarController.swift [hunk 1]
```diff
@@ -15,6 +15,13 @@ class MainTabBarController: UITabBarController, NavigationProtocol {
 
     private lazy var profileTabBarItem = UITabBarItem(title: L10n.profile, image: UIImage(named: "profile_tab"), tag: tabs.firstIndex(of: .profile)!)
 
+
+    /// The viewDidAppear can trigger more than once per lifecycle, setting this flag on the first did appear prevents use from prompting more than once per lifecycle. But still wait until the tab bar has appeared to do so.
+    var viewDidAppear: Bool = false
```
Comment (leandroalonso) [inline podcasts/MainTabBarController.swift]: Maybe `viewDidAppearBefore` is a better name?

**Review by leandroalonso — CHANGES_REQUESTED** [review]
@emilylaguna all the testing steps work as described except for those two:

* Logged out user → Signs In → Eligible for Playback and What's New
* Logged out user → Signs In → Eligible for What's New, not for Playback

What's New for those cases is not shown — and I wonder if it should given it's a fresh install of the app?

File: podcasts/MainTabBarController.swift (hunk 1 above)
Comment (emilylaguna) [reply podcasts/MainTabBarController.swift]: Done here: 4cefe6f0ca5703b3f5cfd58c97091da513fadfed

**Comment by emilylaguna** [comment]
@leandroalonso 
> Logged out user → Signs In → Eligible for Playback and What's New
> Logged out user → Signs In → Eligible for What's New, not for Playback

I updated the steps to remove the what's new part since it shouldn't show in this instance because the app had already been launched before.

**Review by leandroalonso — APPROVED** [review]

**Pull request merged**